include/circt/Support/CPPCDE.h

```diff
@@ -0,0 +1,398 @@
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/ESI/ESIOps.h"
+#include "circt/Dialect/HW/HWOps.h"
+#include "circt/Dialect/HW/HWTypes.h"
+#include "circt/Dialect/Seq/SeqOps.h"
+#include "circt/Support/BackedgeBuilder.h"
+
+#include "assert.h"
+#include <iostream>
+#include <variant>
+
+using namespace mlir;
+using namespace circt;
+
+/*
+CPPCDE: C++ CIRCT Design Entry
+
+Syntactic sugar for creating CIRCT modules and operations. This library is
+intended to be used in places where a large amount of CIRCT IR needs to be
+generated (think generators) within a pass, wherein a syntactically sugar'ed
+approach will lead itself to a significant decrease in the LOC required.
+
+... not a HDL...!
+*/
+
+namespace circt {
+namespace cppcde {
+
+// Port annotations.
+enum class PortKind { Clock, Reset, Default };
+
+// A context defining the operations that define a given semantic.
+// This can be used to overload the inferred operations in a CPPCDE
+// generator.
+struct CDEOperatorSelector {
```

Comment:
Since this meant to be replaced if one has another dialect (e.g. hwarith), should it be called `CDECombOperationSelector`?